File: app/src/main/java/com/example/haidermirza/banner/viper/contract/NormalBannerContract.java
package com.example.haidermirza.banner.viper.contract;

import android.content.Context;

import com.example.haidermirza.banner.adapter.RecyclerViewBannerNormal;
import com.example.haidermirza.banner.library.BannerLayout;

public interface NormalBannerContract {
    interface NB_View{
        void show_message(String str);
    }
    interface NB_Intrector{
        void attach_presenter(NormalBannerContract.NB_Presenter presenter);
        void initial_dataLoad(RecyclerViewBannerNormal banner,RecyclerViewBannerNormal banner2,Context context);
    }
    interface NB_Presenter{
        void attach_view(NormalBannerContract.NB_View view);
        void on_banner_clik(String path);
        void on_initial_dataLoad(RecyclerViewBannerNormal banner,RecyclerViewBannerNormal banner2,Context context);
    }
}

File: app/src/main/java/com/example/haidermirza/banner/viper/contract/NormalBannerContract.kt
package com.example.haidermirza.banner.viper.contract

import android.content.Context
import com.example.haidermirza.banner.adapter.RecyclerViewBannerNormal

interface NormalBannerContract {
    interface NB_View {
        fun show_message(str: String)
    }

    interface NB_Intrector {
        fun attach_presenter(presenter: NormalBannerContract.NB_Presenter)
        fun initial_dataLoad(banner: RecyclerViewBannerNormal, banner2: RecyclerViewBannerNormal, context: Context)
    }

    interface NB_Presenter {
        fun attach_view(view: NormalBannerContract.NB_View)
        fun on_banner_clik(path: String)
        fun on_initial_dataLoad(banner: RecyclerViewBannerNormal, banner2: RecyclerViewBannerNormal, context: Context)
    }
}
